controllers/kubeadmbootstrapconfig_controller.go

```diff
@@ -93,6 +98,30 @@ func (r *KubeadmBootstrapConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Res
 		log.Error(err, "failed to get cluster")
 		return ctrl.Result{}, errors.WithStack(err)
 	}
+
+	// If the machine is not Ready, requeue until it's ready
+	if machine.Status.BootstrapReady == nil || *machine.Status.BootstrapReady == false {
+		log.Info("requeuing as machine is not yet ready", "machine-name", machine.Name)
+		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
+	}
+
+	// if the bootstrap data is not nil then it's already been generated and we don't have anything else to do
+	if machine.Spec.Bootstrap.Data != nil {
+		log.Info("Nothing to be done for this machine", "machine-name", machine.Name)
+		return ctrl.Result{}, nil
+	}
+
+	// Generate the cloud-init for controlplane init
```

Comment:
Am I wrong or we should also expect the UserDataGenerator handle also controlpane join and node join?

Reply:
you are 100% correct. 

As stated in the description:

> This PR generates the control-plane-init cloud init script, but it is incomplete.

Reply:
oh sorry, I missed that